db/seedData/generatePGdata.js

```diff
@@ -0,0 +1,106 @@
+const faker = require('faker');
+
+const placeTypes = ['bakery', 'bar', 'cafe', 'restaurant', 'meal_delivery', 'meal_takeaway'];
+
+function generateRestaurantDataBatch(round) {
+  if (round % 100 === 0) {console.log(round);}
+  if (round > 1000) {
+    return Promise.resolve(null);
+  }
+  let data = [];
+  for (var x = 0; x < 10000; x++) {
+    let restId = (10000 * round) + x;
+    let restaurantRow = generateRestaurantObj(restId);
+    data.push(restaurantRow);
+  }
+  return Promise.resolve(data);
+}
+
+function generateRestaurantObj(restaurantId) {
+  return {
+    restaurant_id: restaurantId,
+    name: faker.company.companyName(), 
+    neighborhood: faker.address.streetName(),
+    zagat_rating: faker.finance.amount(0, 5, 1),
+    google_rating: faker.finance.amount(0, 5, 1),
+    review_count: faker.random.number(100),
+    description: faker.lorem.sentence(10),
+    latitude: faker.address.latitude(),
+    longitude: faker.address.longitude(),
+    address: faker.address.streetAddress('###'),
+    website: faker.internet.url(),
+    price_level: faker.random.number(3) + 1,
+    type: placeTypes[faker.random.number(5)]
+  }
+}
+
+function generateNearbyDataBatch(round) {
+  if (round > 6002) {
+    return Promise.resolve(null);
+  }
+  let data = [];
+  let numIds = setNumIds(round);
+  for (var x = 0; x < numIds; x++) {
+    let restId = (round * 1666) + x;
+    for (var y = 1; y < 7; y++) {
+      let nearbyObj = generateNearbyObj(restId, y);
+      data.push(nearbyObj);
+    }
+  }
+  return Promise.resolve(data);
+}
+
+function setNumIds(round){
+  if (round !== 6002) {
+    return 1666;
+  } else {
+    return 668;
+  }
+}
```

Comment:
Might be worth adding a comment here around the ``generateNearbyDataBatch()`` function and ``setNumIds()`` to explain some of the logic. I'm assuming there are some arbitrary conditional round numbers just to create some variation, but wasn't immediately clear to me if there is something more going on here. Similarly for ``setNumPhotos()`` towards the bottom.

Reply:
In generateNearbyDataBatch, I refactored the logic to make it cleaner to read with batches that had round numbers. I was trying to minimize the number of batches needed to create 6 rows for each of the 10M ids, while adhering to the 10k rows per batch constraint. I refactored by reducing the number of restaurant ids per batch, so that each batch processes the same number of restaurant ids. In setNumPhotos, I was creating planned variation in the number of photos generated for the different restaurant ids, to reduce the number of photo rows that needed to be created. I inserted a comment explaining that so that any ambiguity is removed. Great suggestion!